platform/jvm/MacAppNotarizer.kt

```diff
@@ -0,0 +1,128 @@
+package com.vaticle.bazel.distribution.platform.jvm
+
+import com.vaticle.bazel.distribution.platform.jvm.JVMPlatformAssembler.logger
+import com.vaticle.bazel.distribution.platform.jvm.JVMPlatformAssembler.shell
+import com.vaticle.bazel.distribution.platform.jvm.MacAppNotarizer.Args.ALTOOL
+import com.vaticle.bazel.distribution.platform.jvm.MacAppNotarizer.Args.FILE
+import com.vaticle.bazel.distribution.platform.jvm.MacAppNotarizer.Args.NOTARIZATION_INFO
+import com.vaticle.bazel.distribution.platform.jvm.MacAppNotarizer.Args.NOTARIZE_APP
+import com.vaticle.bazel.distribution.platform.jvm.MacAppNotarizer.Args.PASSWORD
+import com.vaticle.bazel.distribution.platform.jvm.MacAppNotarizer.Args.PRIMARY_BUNDLE_ID
+import com.vaticle.bazel.distribution.platform.jvm.MacAppNotarizer.Args.STAPLE
+import com.vaticle.bazel.distribution.platform.jvm.MacAppNotarizer.Args.STAPLER
+import com.vaticle.bazel.distribution.platform.jvm.MacAppNotarizer.Args.USERNAME
+import com.vaticle.bazel.distribution.platform.jvm.MacAppNotarizer.NotarizationInfoResult.Status.APPROVED
+import com.vaticle.bazel.distribution.platform.jvm.MacAppNotarizer.NotarizationInfoResult.Status.PENDING
+import com.vaticle.bazel.distribution.platform.jvm.MacAppNotarizer.NotarizationInfoResult.Status.REJECTED
+import com.vaticle.bazel.distribution.platform.jvm.MacAppNotarizer.StatusPoller.LOG_FILE_URL
+import com.vaticle.bazel.distribution.platform.jvm.MacAppNotarizer.StatusPoller.MAX_RETRIES
+import com.vaticle.bazel.distribution.platform.jvm.MacAppNotarizer.StatusPoller.POLL_INTERVAL_MS
+import com.vaticle.bazel.distribution.platform.jvm.MacAppNotarizer.StatusPoller.STATUS_MESSAGE_PACKAGE_APPROVED
+import java.nio.file.Path
+
+class MacAppNotarizer(private val options: Options.AppleCodeSigning, private val dmgFilename: String, private val dmgPath: Path) {
```

Comment:
Once a MacOS app has been codesigned, it needs to be submitted to Apple's servers for review.

The review process is called _notarization_ and typically takes a few minutes to complete.